fuse_core/CMakeLists.txt

```diff
@@ -1,519 +1,500 @@
-cmake_minimum_required(VERSION 3.0.2)
+cmake_minimum_required(VERSION 3.5)
 project(fuse_core)
 
-set(build_depends
-  fuse_msgs
-  pluginlib
-  roscpp
-)
+# Default to C++17
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CXX_STANDARD_REQUIRED YES)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+find_package(ament_cmake REQUIRED)
+find_package(fuse_msgs REQUIRED)
+find_package(pluginlib REQUIRED)
+find_package(rcl_interfaces REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rclcpp_components REQUIRED)
 
-find_package(catkin REQUIRED COMPONENTS
-  ${build_depends}
-)
-find_package(Boost REQUIRED COMPONENTS serialization)
 find_package(Ceres REQUIRED)
 find_package(Eigen3 REQUIRED)
-find_package(PkgConfig REQUIRED)
-pkg_check_modules(libglog REQUIRED)
-
-catkin_package(
-  INCLUDE_DIRS
-    include
-  LIBRARIES
-    ${PROJECT_NAME}
-  CATKIN_DEPENDS
-    ${build_depends}
-  DEPENDS
-    Boost
-    CERES
-    EIGEN3
-    GLOG
-)
+find_package(glog REQUIRED)  # Used downstream
```

Comment:
What does downstream mean? Is it not a dependency of this package?

Reply:
It's a dependency of `fuse_constraints`, downstream from `fuse_core`. Core itself doesn't use glog, from what I can tell.

My understanding is that users can also define their own constraints (where they are likely to use glog), so finding glog here means that they won't have to explicitly find it.

Reply:
Ah. I would recommend against depending on something higher in the chain for convenience. It can unnecessarily increase the number of things that need to be installed. For example, say it was a build-only dependency downstream. Downstream could instead depend on it in their build process without requiring further downstream packages to install it.

In this specific case I see `glog/logging.h` is included in 3 `.cpp` files in `fuse_constraints`, but I don't see any `glog` API's used. Maybe it can be removed altogether.

Reply:
Unintuitively, the stuff being used from the `glog` library are actually the `CHECK_XX` macros. :grimacing: 

e.g. 

```
  CHECK_GT(A_.rows(), 0);
  CHECK_EQ(A_.cols(), 3);
```

I'll move the find_package invocation to `fuse_constraints`

Reply:
Moved! aa287f03c254134f577e887a6f9173a2d0b05619

Reply:
If memory serves, the glog dependency was added to make Ceres Solver happy. Cmake support for Ceres had issues in Ubuntu Xenial. I'm actually a little surprised glogging is used directly within fuse at all. Those CHECK_XX macros can probably be switched out for asserts, and glogging dropped as direct dependency.